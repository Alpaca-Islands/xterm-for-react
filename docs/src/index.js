import './index.css'

import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import HelloWorld from './examples/HelloWorld'
import Echo from './examples/Echo'
import Addons from './examples/Addons'

import {
    BrowserRouter as Router,
    Switch,
    Route,
  } from "react-router-dom";

const routing = (
    <Router>
        <Switch>
            <Route exact path="/" component={App} />
            <Route path="/hello-world" component={HelloWorld} />
            <Route path="/echo" component={Echo} />
            <Route path="/addons" component={Addons} />
        </Switch>
    </Router>
)

ReactDOM.render(routing, document.getElementById('root'))