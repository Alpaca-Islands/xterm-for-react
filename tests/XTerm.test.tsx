import { mount } from 'enzyme'
import * as React from 'react'
import { XTerm } from '../src'

describe('<XTerm />', () => {
	it('should render correctly', () => {
		const wrapper = mount(<XTerm />)
		expect(wrapper.html()).toMatchSnapshot()
	})
})
