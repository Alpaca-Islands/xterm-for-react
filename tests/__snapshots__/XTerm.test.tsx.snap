// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XTerm /> should render correctly 1`] = `"<div><div dir=\\"ltr\\" class=\\"terminal xterm\\" tabindex=\\"0\\"><div class=\\"xterm-viewport\\" style=\\"background-color: rgb(0, 0, 0);\\"><div class=\\"xterm-scroll-area\\"></div></div><div class=\\"xterm-screen\\"><div class=\\"xterm-helpers\\"><textarea class=\\"xterm-helper-textarea\\" aria-label=\\"Terminal input\\" aria-multiline=\\"false\\" autocorrect=\\"off\\" autocapitalize=\\"off\\" spellcheck=\\"false\\" tabindex=\\"0\\"></textarea><span class=\\"xterm-char-measure-element\\" aria-hidden=\\"true\\" style=\\"font-family: courier-new, courier, monospace; font-size: 15px;\\">W</span><div class=\\"composition-view\\"></div></div><canvas class=\\"xterm-text-layer\\" style=\\"z-index: 0;\\"></canvas><canvas class=\\"xterm-selection-layer\\" style=\\"z-index: 1;\\"></canvas><canvas class=\\"xterm-link-layer\\" style=\\"z-index: 2;\\"></canvas><canvas class=\\"xterm-cursor-layer\\" style=\\"z-index: 3;\\"></canvas></div></div></div>"`;
